pkg/storegateway/bucket_stores.go

```diff
@@ -551,17 +560,39 @@ func (u *BucketStores) closeBucketStoreAndDeleteLocalFilesForExcludedTenants(inc
 	}
 }
 
-// getBlocksLoadedMetric returns the number of blocks currently loaded across all bucket stores.
-func (u *BucketStores) getBlocksLoadedMetric() float64 {
-	count := 0
+// countBlocksLoaded returns the total number of blocks loaded and the number of blocks
+// loaded bucketed by the configured block durations, summed for all users.
+func (u *BucketStores) countBlocksLoaded() (int, map[time.Duration]int) {
+	byDuration := make(map[time.Duration]int)
+	total := 0
 
 	u.storesMu.RLock()
+	defer u.storesMu.RUnlock()
+
 	for _, store := range u.stores {
-		count += store.Stats().BlocksLoaded
+		stats := store.Stats(u.cfg.TSDB.BlockRanges)
```

Comment:
`u.cfg.TSDB` was a config supposed to be used only by the ingester. This means that it's supposed to only have 1 range (the 2h one). I think what you want here are the compactor ranges, not the ingester ones. However, reading a compactor config looks a bad small to me, so I'm wondering if we could just keep it easy and have hardcoded buckets instead.

Reply:
I can make that change. Is your preference for using the compactor config or hard-coding the expected buckets?